### ISSUE TITLE
awsmobile push fails on react-native project "no such file or directory"

### ISSUE THREAD

**Issue by JohnRuffin** [aws/awsmobile-cli#23]
My steps:
- Created a react-native project with `create-react-native-app projectName`

- `cd projectName && awsmobile init`

- `npm install --save aws-amplify-react-native`

Added some analytics events and I can see them in the AWS console.  **No problem**.

However, when I do a `awsmobile push` I get the following "no such file or directory" error on _awsmobilejs/.awsmobile/backend-build/mobile-hub-project.yml_'
```
building backend
   generating backend project content
/Users/kingofcode/myDev/sources/awsmobile-cli/node_modules/aws-sdk/lib/request.js:31
            throw err;
            ^
Error: ENOENT: no such file or directory, open '/Users/kingofcode/myDev/AwesomeWithAWSMobile/awsmobilejs/.awsmobile/backend-build/mobile-hub-project.yml'
    at Object.fs.openSync (fs.js:652:18)
    at Object.fs.writeFileSync (fs.js:1299:33)
    at Object.dumpYaml (/Users/kingofcode/myDev/sources/awsmobile-cli/lib/aws-operations/mobile-yaml-ops.js:42:8)
    at Object.generateContents (/Users/kingofcode/myDev/sources/awsmobile-cli/lib/aws-operations/mobile-api-content-generator.js:27:22)
    at Object.build (/Users/kingofcode/myDev/sources/awsmobile-cli/lib/backend-operations/ops-project.js:30:31)
    at /Users/kingofcode/myDev/sources/awsmobile-cli/lib/project-backend-builder.js:44:40
    at Object.build (/Users/kingofcode/myDev/sources/awsmobile-cli/lib/backend-operations/ops-user-signin.js:69:9)
    at /Users/kingofcode/myDev/sources/awsmobile-cli/lib/project-backend-builder.js:41:32
    at Array.forEach (native)
    at Object.build (/Users/kingofcode/myDev/sources/awsmobile-cli/lib/project-backend-builder.js:39:33)
```

[**Root cause analysis**] The problem is with this line in **mobile-yaml-ops.js**: `fs.writeFileSync(ymlFilePath, yml, 'utf8')` as it tries to write to a file _**but the directory doesn't exist**_.  

[**Solution**] I've fixed this in my fork and will submit a PR.

**Comment by mlabieniec** [comment]
Fixed in PR #22 pending npm publish

**Comment by mlabieniec** [comment]
Fixed and published in 1.0.6
